core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3322,49 +3363,69 @@ private SyncResult syncInodeMetadata(RpcContext rpcContext, LockedInodePath inod
         }
       }
       if (syncPlan.toLoadMetadata()) {
-        loadMetadata = true;
+        AlluxioURI mountUri = new AlluxioURI(mMountTable.getMountPoint(inodePath.getUri()));
+        // We only sync up to the mount point or the originalPath, whichever is lower in the tree
+        if (PathUtils.hasPrefix(originalPath.getUri().getPath(), mountUri.getPath())) {
+          pathsToLoad.add(originalPath.getUri().getPath());
+        } else {
+          pathsToLoad.add(mountUri.getPath());
+        }
       }
       if (syncPlan.toSyncChildren() && inode instanceof InodeDirectory
           && syncDescendantType != DescendantType.NONE) {
         UfsStatus[] listStatus = ufs.listStatus(ufsUri.toString());
+
+        InodeDirectory inodeDir = (InodeDirectory) inode;
+        // maps children name to inode
+        Map<String, Inode<?>> inodeChildren = new HashMap<>();
+        for (Inode<?> child : inodeDir.getChildren()) {
+          inodeChildren.put(child.getName(), child);
+        }
+
+        // if the alluxio path contains a mount point, we need to keep going to find it
```

Comment:
I think this comment is not relevant to this code block anymore?

Reply:
done